app/src/main/java/com/example/pinterbest/LoginFragment.kt

```diff
@@ -24,30 +36,127 @@ class LoginFragment : Fragment() {
             container,
             false
         )
+
+        sessionManager = SessionRepository(
+            preferences = requireActivity().getSharedPreferences(
+                getString(R.string.login_info),
+                Context.MODE_PRIVATE
+            )
+        )
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         binding.loginButton.setOnClickListener {
-            if (isValidLogIn()) {
-                Toast.makeText(context, "Correct", Toast.LENGTH_SHORT).show()
-                it.findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
+            if (validateUserFields()) {
+                val model = ViewModelProvider(
+                    requireActivity(),
+                    LoginFactory(
+                        requireActivity().application,
+                        getUserData(),
+                        Repository(
+                            preferences = requireActivity().getSharedPreferences(
+                                getString(R.string.login_info),
+                                Context.MODE_PRIVATE
+                            )
+                        )
+                    )
+                ).get(LoginViewModel::class.java)
+
+                model.logInCodeLiveData.observe(viewLifecycleOwner) { response ->
+                    loginProvider(view, response)
+                }
             } else {
-                Toast.makeText(context, "Not correct", Toast.LENGTH_SHORT).show()
-                it.findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
+                Toast.makeText(context, "Произошла ошибка сети!", Toast.LENGTH_SHORT).show()
             }
         }
+
+        binding.emailAuth.setOnClickListener {
+            it.findNavController().navigate(R.id.action_loginFragment_to_registrationFragment)
+        }
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
-    private fun isValidLogIn(): Boolean {
-        return Validator.isValidName(binding.usernameBox.text.toString().trim()) &&
-            Validator.isValidPassword(binding.passwordBox.text.toString().trim())
+    private fun setUpBottomNavigationItem() {
+        (activity as MainActivity).binding.bottomNavigation.menu
+            .getItem(MainActivity.HOME_POSITION_BNV).isChecked = true
+    }
+
+    private fun loginProvider(view: View, response: Response<ResponseBody>) {
```

Comment:
actually, `Response` is a data layer element. It shouldn't be used on UI level
I suggest you to operate with exceptions to implement the following logic. But mapping response.code to exception or data move to Repository (data layer)

Reply:
Created AuthRepository